parser/parser.go

```diff
@@ -245,6 +250,64 @@ func parseUnary(p *Parser) (ast.Node, error) {
 	return ast.NewUnaryExpr(tok.Type, expr), nil
 }
 
+func parseVarDecls(p *Parser) (ast.Node, error) {
+	p.forget(1)
+	return _parseVarDecls(p)
+}
+
+func _parseVarDecls(p *Parser) (ast.VarDecls, error) {
+
+	if !p.scry(2) {
```

Comment:
Why lookahead of 2?

var a; // no lookahead
var a = 1; // no lookahead

Reply:
At this point I had just "var". I dropped it since it is not necessary anymore. Now I need at least two tokens to see if it is:

* identifier + semicolon
* identifier + assignment (=)

How could this be written differently ?

Reply:
```go
ident := p.next()
if ident.Type() != token.Ident {
	return nil, p.errorf(ident, "expected IDENT")
}

tok := p.next()
if tok.Type() == token.Semicolon {
	return ast.NewVarDecls(ast.NewVarDecl(ident.Value, ast.NewUndefined())), nil
}

if tok.Type != token.Assign {
	return nil, fmt.Errorf("parser: var decl: expected assignment token [=] got [%s]", tok)
}

// parse assignment ...
```